SimPPS/PPSPixelDigiProducer/src/RPixChargeShare.cc

```diff
@@ -0,0 +1,162 @@
+      if (pixel_row == 0 || pixel_row == pxlRowSize_ - 1)
+        pixel_width_x = 0.1;  // Correct edge pixel width
+      if (pixel_col == 0 || pixel_col == pxlColSize_ - 1)
+        pixel_width_y = 0.15;  //
+      double pixel_center_x = pixel_lower_x + (pixel_width_x) / 2.;
+      double pixel_center_y = pixel_lower_y + (pixel_width_y) / 2.;
+      // xbin and ybin are coordinates (um) ??nside the pixel as in the test beam, swapped wrt plane coodinates.
+      int xbin = int((((*i).Y() - pixel_center_y) + pixel_width_y / 2.) * 1.e3 / 5.);
+      int ybin = int((((*i).X() - pixel_center_x) + pixel_width_x / 2.) * 1.e3 / 5.);
+      if (pixel_width_x < 0.11 && pixel_width_y < 0.151) {  // pixel 100x150 um^2
+        psize = 0;
+        if (xbin > xBinMax[psize] || ybin > yBinMax[psize])
+          continue;
+      }
+      if (pixel_width_x > 0.11 && pixel_width_y < 0.151) {  // pixel 200x150 um^2
+        psize = 2;
+        if (xbin > xBinMax[psize] || ybin > yBinMax[psize])
+          continue;
+      }
+      if (pixel_width_x < 0.11 && pixel_width_y > 0.151) {  // pixel 100x300 um^2
+        psize = 1;
+        if (xbin > xBinMax[psize] || ybin > yBinMax[psize])
+          continue;
+      }
+      if (pixel_width_x > 0.11 && pixel_width_y > 0.151) {  // pixel 200x300 um^2
+        psize = 3;
+        if (xbin > xBinMax[psize] || ybin > yBinMax[psize])
+          continue;
+      }
+      double hit2neighbour[8];
+      double collect_prob = chargeMap2E[psize][xbin][ybin];
+      thePixelChargeMap[pixel_no] += charge_in_pixel * collect_prob;
+      unsigned short neighbour_no[8];
+      unsigned short m = 0;
+      double closer_neighbour = 0;
+      unsigned short closer_no = 0;
+      //      Considering the 8 neighbours to share charge
+      for (int k = pixel_row - 1; k <= pixel_row + 1; k++) {
+        for (int l = pixel_col - 1; l <= pixel_col + 1; l++) {
+          if ((k < 0) || k > pxlRowSize_ - 1 || l < 0 || l > pxlColSize_ - 1)
+            continue;
+          if ((k == pixel_row) && (l == pixel_col))
+            continue;
+          double neighbour_pixel_lower_x = 0;
+          double neighbour_pixel_lower_y = 0;
+          double neighbour_pixel_upper_x = 0;
+          double neighbour_pixel_upper_y = 0;
+          double neighbour_pixel_center_x = 0;
+          double neighbour_pixel_center_y = 0;
+          //      Check the hit approach to the neighbours
+          theRPixDetTopology_.pixelRange(
+              k, l, neighbour_pixel_lower_x, neighbour_pixel_upper_x, neighbour_pixel_lower_y, neighbour_pixel_upper_y);
+          neighbour_pixel_center_x = neighbour_pixel_lower_x + (neighbour_pixel_upper_x - neighbour_pixel_lower_x) / 2.;
+          neighbour_pixel_center_y = neighbour_pixel_lower_y + (neighbour_pixel_upper_y - neighbour_pixel_lower_y) / 2.;
+          hit2neighbour[m] =
+              1 / sqrt(pow((*i).X() - neighbour_pixel_center_x, 2.) + pow((*i).Y() - neighbour_pixel_center_y, 2.));
```

Comment:
@mundim , better avoid using pow() for power 2 - in some compilers it is transformed to right thing but at the same time easy to write a code without pow in this case.

Reply:
What about changing 1/pow(x,2) to pow(x,-2). Would that comply with the rules?

Reply:
this suggestion was motivated by an old technique of code optimization, where the inverse is better implemented with negative power.